orttraining/orttraining/test/training_ops/cpu/nn/dropout_op_test.cc

```diff
@@ -168,16 +168,25 @@ void RunDropoutGradTest(const char* op, float ratio, const std::vector<int64_t>&
   const float output_constant = input_constant / (1.0f - ratio);
   std::vector<float> dx_data{};
   dx_data.reserve(input_shape.Size());
-  std::transform(
-      mask_buffer.get(), mask_buffer.get() + input_shape.Size(), std::back_inserter(dx_data),
-      [output_constant](bool mask_value) { return mask_value ? output_constant : 0.0f; });
+
+  if (training_mode) {
+    std::transform(
+        mask_buffer.get(), mask_buffer.get() + input_shape.Size(), std::back_inserter(dx_data),
+        [output_constant](bool mask_value) { return mask_value ? output_constant : 0.0f; });
+  } else {
+    dx_data.resize(input_shape.Size(), input_constant);
```

Comment:
vector::assign() seems more in line with the intent. resize() may keep existing elements (even though there aren't any).

Reply:
good catch